src/codegen/transformations.lisp

```diff
@@ -246,3 +201,169 @@
     (fset:do-map (name entry (immutable-map-data (tc:environment-function-environment env)))
       (setf (gethash name table) (tc:function-env-entry-arity entry)))
     table))
+
+(defvar binding-list-traversals
```

Comment:
This is a list of the five cases which got special treatment inside of `traverse` previously, where `bound-variables` gets updated and passed, instead of passed through the recursion with no change. Is there a better way of doing this than `defvar`? (Lisp seemed to not like it when I tried to define it as a constant...) Should I inline this into `traverse-binding-list` below?

Reply:
```
(defun make-binding-list-traversals ()
    (load-time-value
      (list ...)
      t))
```